biz.aQute.repository/src/aQute/bnd/repository/p2/provider/P2Indexer.java

```diff
@@ -130,12 +132,29 @@ private Repository readRepository() throws Exception {
 		P2Impl p2 = new P2Impl(client, this.url, Processor.getExecutor());
 		List<Artifact> artifacts = p2.getArtifacts();
 		List<Promise<Resource>> fetched = new ArrayList<>(artifacts.size());
-		Set<URI> visited = new HashSet<>(artifacts.size());
+		Set<BundleId> visited = new HashSet<>(artifacts.size());
+
+		final Map<BundleId,Resource> existing = new HashMap<>();
+		if (getBridge() != null) {
```

Comment:
Yes I agree, I have been looking at adding md5 support and it seems feasible with (I hope) fairly minimal additional complexity. It would be so much easier if p2 already supported sha-256! But I think this is probably a fair ways off.